compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -637,6 +642,29 @@ pub struct BoundVarReplacer<'me, 'tcx> {
     universe_indices: &'me mut Vec<Option<ty::UniverseIndex>>,
 }
 
+pub fn with_replaced_escaping_bound_vars<'a, 'tcx, T: TypeFoldable<'tcx>, R: TypeFoldable<'tcx>>(
```

Comment:
I don't know how I feel about making this pattern more common. This probably deserves to be split into its own change with its own justification.

Reply:
Can you add a comment and maybe a FIXME that we should be more careful to not leak placeholders?